components/TemplateGrid/TemplateGridItemArea.vue

```diff
@@ -23,13 +27,44 @@ export default {
             type: Number,
             required: true,
         },
+        rowsHeight: {
+            type: Number,
+            required: true,
+        },
+        columnWidth: {
+            type: Number,
+            required: true,
+        },
     },
+    data: () => ({
+        gridGap: 10,
+    }),
     computed: {
+        ...mapState('tree', {
+            treeData: state => state.treeData,
+        }),
         gridItemStyles() {
-            const { ghost, column } = this.item;
+            const { column } = this.item;
             return {
                 gridColumn: `${column + 1} / ${this.columns + 1}`,
-                padding: ghost ? '0 1px 1px' : '5px',
+                padding: `8px 0 8px ${this.gridGap}px`,
+                gridTemplateColumns: `repeat(${this.columns - column}, 1fr)`,
+                gridGap: `${this.gridGap}px`,
+            };
+        },
+        setConnectionStyle() {
```

Comment:
Not efficient. Whenever this.item mutate it will recalculate everything once again, it seems like too many things are inside computed. Would not it better to manipulate transforms instead of positioning it via fixed positions ?